### PR TITLE
Update support for AWS security group

### PR DESCRIPTION
**Pull request by ayalganem**
- Update support for AWS security group

```
=== RUN   TestAccDataSourceCloudSecurityGroupAWSBasic
--- PASS: TestAccDataSourceCloudSecurityGroupAWSBasic (5.62s)
=== RUN   TestAccDataSourceCloudAccountAWSBasic
--- PASS: TestAccDataSourceCloudAccountAWSBasic (3.89s)
=== RUN   TestAccDataSourceCloudAccountAzureBasic
--- PASS: TestAccDataSourceCloudAccountAzureBasic (2.97s)
=== RUN   TestAccDataSourceCloudAccountGCPBasic
--- PASS: TestAccDataSourceCloudAccountGCPBasic (3.60s)
=== RUN   TestAccDataSourceContinuousComplianceNotificationBasic
--- PASS: TestAccDataSourceContinuousComplianceNotificationBasic (1.77s)
=== RUN   TestAccDataSourceContinuousCompliancePolicyBasic
--- PASS: TestAccDataSourceContinuousCompliancePolicyBasic (4.30s)
=== RUN   TestAccDataSourceIpListBasic
--- PASS: TestAccDataSourceIpListBasic (2.03s)
=== RUN   TestAccDataSourceRoleBasic
--- PASS: TestAccDataSourceRoleBasic (1.73s)
=== RUN   TestAccDataSourceRuleSetBasic
--- PASS: TestAccDataSourceRuleSetBasic (2.51s)
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestAccResourceCloudSecurityGroupAWSBasic
--- PASS: TestAccResourceCloudSecurityGroupAWSBasic (6.02s)
=== RUN   TestAccResourceCloudAccountAWSBasic
--- PASS: TestAccResourceCloudAccountAWSBasic (6.45s)
=== RUN   TestAccResourceCloudAccountAzureBasic
--- PASS: TestAccResourceCloudAccountAzureBasic (4.72s)
=== RUN   TestAccResourceCloudAccountGCPBasic
--- PASS: TestAccResourceCloudAccountGCPBasic (5.80s)
=== RUN   TestAccResourceContinuousComplianceNotificationBasic
--- PASS: TestAccResourceContinuousComplianceNotificationBasic (3.44s)
=== RUN   TestAccResourceContinuousCompliancePolicyBasic
--- PASS: TestAccResourceContinuousCompliancePolicyBasic (6.93s)
=== RUN   TestAccResourceIPListBasic
--- PASS: TestAccResourceIPListBasic (3.85s)
=== RUN   TestAccResourceRoleBasic
--- PASS: TestAccResourceRoleBasic (3.22s)
=== RUN   TestAccResourceRuleSetBasic
--- PASS: TestAccResourceRuleSetBasic (3.75s)
PASS
```

### CODE REVIEW

File: dome9/resource_dome9_cloud_security_group_aws.go [hunk 1]
```diff
@@ -260,6 +251,44 @@ func resourceCloudSecurityGroupAWSDelete(d *schema.ResourceData, meta interface{
 	return nil
 }
 
+func resourceCloudSecurityGroupAWSUpdate(d *schema.ResourceData, meta interface{}) error {
+	d9Client := meta.(*Client)
+	if d.HasChange("is_protected") {
+		protectionMode := getProtectionMode(d.Get("is_protected").(bool))
+		log.Printf("[INFO] Updating security group protection mode to: %s", protectionMode)
+
```
Comment (erezweinstein5-zz) [inline dome9/resource_dome9_cloud_security_group_aws.go]: no need for blank lines after log

File: dome9/resource_dome9_cloud_security_group_aws_test.go [hunk 2]
```diff
@@ -32,12 +32,21 @@ func TestAccResourceCloudSecurityGroupAWSBasic(t *testing.T) {
...
 					testAccCheckCloudSecurityGroupAWSExists(securityGroupTypeAndName, &cloudSecurityGroupAWSResponse),
 					resource.TestCheckResourceAttr(securityGroupTypeAndName, "dome9_security_group_name", variable.AWSSecurityGroupName),
 					resource.TestCheckResourceAttr(securityGroupTypeAndName, "description", variable.AWSSecurityGroupDescription),
 					resource.TestCheckResourceAttr(securityGroupTypeAndName, "aws_region_id", variable.AWSSecurityGroupRegionID),
+					resource.TestCheckResourceAttr(securityGroupTypeAndName, "tags.tag_key", variable.AWSSecurityGroupTagValue),
+				),
+			},
+
+			{
+				Config: testAccCheckCloudSecurityGroupAWSBasic(awsCloudAccountHCL, awsTypeAndName, securityGroupGeneratedName, securityGroupTypeAndName, variable.AWSSecurityGroupUpdateTagValue),
```
Comment (erezweinstein5-zz) [inline dome9/resource_dome9_cloud_security_group_aws_test.go:46]: add a little note that this part is testing update tag

File: dome9/resource_dome9_cloud_security_group_aws_test.go [hunk 3]
```diff
@@ -113,6 +126,7 @@ data "%s" "%s" {
 		variable.AWSSecurityGroupDescription,
 		variable.AWSSecurityGroupRegionID,
 		awsCloudAccountTypeAndName,
+		tagVale,
```
Comment (erezweinstein5-zz) [inline dome9/resource_dome9_cloud_security_group_aws_test.go]: typo

**Review by erezweinstein5-zz — CHANGES_REQUESTED** [review]

File: dome9/resource_dome9_cloud_security_group_aws.go [hunk 4]
```diff
@@ -260,6 +251,44 @@ func resourceCloudSecurityGroupAWSDelete(d *schema.ResourceData, meta interface{
...
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getProtectionMode(isProtected bool) (protectionMode string) {
+	if isProtected {
+		protectionMode = "FullManage"
```
Comment (uditamar) [inline dome9/resource_dome9_cloud_security_group_aws.go]: this must be set out of const

**Pull request merged**